src/shims/unix/sync.rs

```diff
@@ -124,79 +124,40 @@ fn mutex_get_id<'tcx>(
         |ecx| {
             // This is called if a static initializer was used and the lock has not been assigned
             // an ID yet. We have to determine the mutex kind from the static initializer.
-            let mutex = ecx.deref_pointer(mutex_op)?.ptr();
-            let contents = get_contents(ecx, mutex)?;
-            let kind = kind_from_static_initializer(ecx, contents)?;
+            let kind = kind_from_static_initializer(ecx, mutex_op)?;
 
             Ok(Some(AdditionalMutexData { kind, address }))
         },
     )?;
-    // mutex_check_moved(ecx, mutex_op, id)?;
+
+    // Check that the mutex has not been moved since last use.
     let data = ecx.mutex_get_data(id).expect("data should be always exist for pthreads");
     if data.address != address {
         throw_ub_format!("pthread_mutex_t can't be moved after first use")
     }
+
     Ok(id)
 }
 
-/// Returns the kind of a mutex, based on its contents.
-/// The contents must not contain the initial "header" where the mutex id is stored.
+/// Returns the kind of a static initializer.
 fn kind_from_static_initializer<'tcx>(
     ecx: &MiriInterpCx<'tcx>,
-    contents: &[u8],
+    mutex_op: &OpTy<'tcx>,
 ) -> InterpResult<'tcx, MutexKind> {
-    let on_linux = &*ecx.tcx.sess.target.os == "linux";
-
-    let default_initializer =
-        ecx.eval_path(&["libc", "PTHREAD_MUTEX_INITIALIZER"]).assert_mem_place().ptr();
-    if contents == get_contents(ecx, default_initializer)? {
+    // Only linux has static initializers other than PTHREAD_MUTEX_DEFAULT.
+    if &*ecx.tcx.sess.target.os != "linux" {
```

Comment:
Please make this an exhaustive match, like the old `mutex_get_kind`. When we add a new target we need to check that this is still true.